odilia/src/tower.rs

```diff
@@ -27,17 +33,71 @@ type Response = Vec<Command>;
 type Request = Event;
 type Error = OdiliaError;
 
+pub struct SerialFutures<'a, T, O> {
```

Comment:
from [stackoverflow](https://stackoverflow.com/questions/48081740/executing-a-collection-of-futures-sequentially)

you can directly use a stream of streams of future, where each future is converted to a stream and then flattened to force it to be poled to completion before anything else is. For that, you can use iter_ok, like so...

```rust
iter_ok(tasks).map(|f|f.into_stream()).flatten()
```

or this, where you get a single future which poles all of them to completion

```rust
let tasks = vec![ok(()), ok(()), ok(())];

let combined_task = iter_ok::<_, ()>(tasks).for_each(|f| f);
```

though I still question the doing of such serially, but this is an approach with mucbh less code, if it can be done

Reply:
I like this! But we need concrete types since we need to return this from a function.

Reply:
I tried really hard to use something more similar to this, especially in relation to not rolling our own `SequentialFutures`. That does seem silly after all.

I got really close but got slapped by the borrow checker. I can come back to this one.